### PR TITLE
Upgrade Evercrypt code

### PR DESCRIPTION
**Pull request by wintersteiger**
This is the latest Evercrypt code. I made a few minor changes; this corresponds to https://github.com/project-everest/hacl-star/pull/345.

### CODE REVIEW

**Comment by ashamis** [comment]
Can you please run the daily tests. It would be good to have a run with the sanitizer.

**Comment by achamayou** [comment]
@wintersteiger to trigger the daily build, the easiest thing to do is to do a small edit to .daily.yml (like add a line return).

Was anything under 3rdparty manually edited, or did it all come from the script and only other paths needed updates?

**Comment by achamayou** [comment]
@wintersteiger Could you also update the sha in cgmanifest.json for https://github.com/project-everest/hacl-star please?

File: cmake/crypto.cmake [hunk 1]
```diff
@@ -24,7 +24,7 @@ if("sgx" IN_LIST COMPILE_TARGETS)
   )
   target_compile_definitions(
     evercrypt.enclave PRIVATE INSIDE_ENCLAVE KRML_HOST_PRINTF=oe_printf
-                              KRML_HOST_EXIT=oe_abort
+                              KRML_HOST_EXIT=oe_abort LINUX_NO_EXPLICIT_BZERO
```
Comment (achamayou) [inline cmake/crypto.cmake:27]: What does that flag do?

File: src/node/history.h [hunk 2]
```diff
@@ -208,20 +208,21 @@ namespace ccf
   private:
     uint64_t index;
     uint32_t max_index;
+    static const uint32_t hash_size = 32U;
```
Comment (achamayou) [inline src/node/history.h:211]: I don't think we need that, that's just Sha256Hash::SIZE.

**Review by achamayou — APPROVED** [review]

File: cmake/crypto.cmake (hunk 1 above)
Comment (wintersteiger) [reply cmake/crypto.cmake:27]: This tells the build that there's no [explicit_bzero](https://www.freebsd.org/cgi/man.cgi?query=explicit_bzero&sektion=3). I don't know what exactly that's used for though.

File: src/node/history.h (hunk 2 above)
Comment (wintersteiger) [reply src/node/history.h]: That's right, done.

**Comment by wintersteiger** [comment]
@achamayou no modifications, all of those have gone into hacl-star before updating the files. `cgmanifest` updated.

File: cmake/crypto.cmake (hunk 1 above)
Comment (achamayou) [reply cmake/crypto.cmake:27]: Huh. There doesn't seem to be any calls to `bzero()` in the evercrypt code. musl as provided in openenclave does expose both `bzero()` and `explicit_bzero()` though, so should we have this flag at all?

Comment (wintersteiger) [reply cmake/crypto.cmake:27]: I see this:

    root@45ddd2ed032b:/CCF/3rdparty/hacl-star/evercrypt# grep bzero *
    Lib_Memzero0.c:    explicit_bzero(dst, len_);

If it's properly supported by musl we may want to update the macro-checks in `Lib_Memzero0.c` (the build explicitly asks for pull requests...)

**Comment by wintersteiger** [comment]
Done! This is ready to merge once the CI finishes.

**Comment by wintersteiger** [comment]
Hmm:

    /opt/oe_lvi/ld: libevercrypt.enclave.a(Lib_Memzero0.c.o): in function `Lib_Memzero0_memzero':
    /CCF/3rdparty/hacl-star/evercrypt/Lib_Memzero0.c:40: undefined reference to `explicit_bzero'

Is this special to the LVI mitigations and do we have to compile them with `_GNU_SOURCE` too to get access to `explicit_bzero`?

**Comment by wintersteiger** [comment]
It looks like OE's musl doesn't have `explicit_bzero`, I can't find it in the .a. I'm falling back to `LINUX_NO_EXPLICIT_BZERO` so we can merge this while I search for a substitute.

**Pull request merged**